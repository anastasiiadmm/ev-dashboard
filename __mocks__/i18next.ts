import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import translationEN from '~/shared/locales/translationEN.json';
import translationRU from '~/shared/locales/translationRU.json';
import translationKY from '~/shared/locales/translationKY.json';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: { translation: translationEN },
      ru: { translation: translationRU },
      ky: { translation: translationKY },
    },
    lng: 'ru',
    fallbackLng: 'ru',
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
