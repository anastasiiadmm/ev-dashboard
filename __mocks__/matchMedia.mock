window.matchMedia = window.matchMedia || function(query) {
  return {
    matches: false,
    addListener: function() {},
    removeListener: function() {},
    media: query,
    onchange: null,
    dispatchEvent: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  };
};
