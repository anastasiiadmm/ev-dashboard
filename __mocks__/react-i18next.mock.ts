jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
  Trans: ({ children }) => children,
}));
