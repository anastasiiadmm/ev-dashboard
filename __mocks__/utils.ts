jest.mock('mobx', () => ({
  ...jest.requireActual('mobx'),
  observable: jest.fn(() => ({})),
}));
