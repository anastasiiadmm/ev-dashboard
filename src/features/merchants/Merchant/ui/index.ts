export { default as CardMerchantHeader } from './CardMerchantHeared/CardMerchantHeader';
export { default as CardMerchantInfo } from './CardMerchantInfo/CardMerchantInfo';
export { default as TextBlock } from './TextBlock/TextBlock';
