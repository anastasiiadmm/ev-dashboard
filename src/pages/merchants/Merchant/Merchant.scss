@use '../../../app/styles/variables' as variables;

.Merchant {
  padding: 28px 20px;

  &__card-text-block {
    > div {
      display: flex;
      flex-direction: column;
    }
  }

  &__title-info-block {
    margin: 0 0 5px;
    font-size: 24px;
    line-height: 36px;
  }

  &__title-info {
    margin: 0 0 5px;
    font-weight: 600;
    line-height: 18px;
    text-transform: uppercase;
  }

  &__text-info-block {
    font-size: 14px;
    font-weight: 500;
    color: #707a94;
  }

  &__button-style {
    display: flex;
    align-items: center;
    height: 34px;
    align-self: self-end;
    padding: 8px 32px 8px 32px;
  }

  .ant-card,
  .ant-tabs {
    width: 100%;
  }

  .ant-tabs-top > .ant-tabs-nav {
    margin: 1px 0 16px !important;
  }

  .ant-tabs .ant-tabs-tab {
    padding: 8px 10px !important;
  }
}
