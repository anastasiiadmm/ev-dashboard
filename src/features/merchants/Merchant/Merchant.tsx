import React from 'react';
import { Row, Tabs } from 'antd';
import bem from 'easy-bem';
import dayjs from 'dayjs';

import { IMerchantInfo, ITabs } from '~/features/merchants/interfaces';
import { CardMerchantHeader, CardMerchantInfo } from '~/features/merchants/Merchant/ui';
import './Merchant.scss';

const merchantItems: IMerchantInfo = {
  id: 64,
  number_stations: 33,
  active_stations: 20,
  inactive_stations: 13,
  name: 'We way',
  legal_name: 'ИП Ким ЛВ',
  email: 'weway@mail.com',
  rate: '64',
  agreement_number: 'W16/09/2023',
  address: 'Манаса 5',
  phone: '+996477011',
  active: true,
  country: 'Кыргызстан',
  district: 'Первомайский',
  city: 'Бишкек',
  created_by: 'joe@mail.com',
  created_at: dayjs().format(),
};

const Merchant = () => {
  const b = bem('Merchant');

  const items: ITabs[] = [
    {
      key: '1',
      label: 'О мерчанте',
      children: (
        <CardMerchantInfo
          merchant={merchantItems}
          classNameButton={b('button-style')}
          classNameTitle={b('title-info')}
        />
      ),
    },
    {
      key: '2',
      label: 'Станции',
      children: 'Список станций',
    },
  ];

  return (
    <Row justify='space-between' data-testid='auth-component' className={b()}>
      <CardMerchantHeader className={b('card-text-block')} merchant={merchantItems} />
      <Tabs defaultActiveKey='1' items={items} />
    </Row>
  );
};

export default Merchant;
