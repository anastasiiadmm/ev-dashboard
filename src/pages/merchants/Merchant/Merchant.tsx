import React, { useEffect } from 'react';
import { Row, Spin, Tabs } from 'antd';
import bem from 'easy-bem';
import { toJS } from 'mobx';
import { useParams } from 'react-router';
import { observer } from 'mobx-react-lite';
import { useTranslation } from 'react-i18next';

import { CardMerchantHeader, CardMerchantInfo, TableStations } from '~/pages/merchants/Merchant/ui';
import { ITabs } from '~/pages/merchants/interfaces';
import { merchantStore } from '~/shared/api/store';
import { useLanguage } from '~/shared/context';
import './Merchant.scss';

const Merchant = observer(() => {
  const b = bem('Merchant');
  const { t } = useTranslation();
  const { id } = useParams();
  const { currentLanguage } = useLanguage();
  const { merchantDetail, merchantDetailLoading } = toJS(merchantStore);

  useEffect(() => {
    merchantStore.getMerchantDetail(currentLanguage, Number(id));
  }, [currentLanguage, id]);

  const items: ITabs[] = [
    {
      key: '1',
      label: t('merchants.about_merchant') as string,
      children: (
        <CardMerchantInfo
          merchant={merchantDetail}
          classNameButton={b('button-style')}
          classNameTitle={b('title-info')}
        />
      ),
    },
    {
      key: '2',
      label: t('merchants.stations') as string,
      children: <TableStations />,
    },
  ];

  if (merchantDetailLoading) {
    return <Spin className='spin' />;
  }

  return (
    <Row justify='space-between' data-testid='merchant-component' className={b()}>
      <CardMerchantHeader className={b('card-text-block')} merchant={merchantDetail} />
      <Tabs defaultActiveKey='1' items={items} />
    </Row>
  );
});

export default Merchant;
