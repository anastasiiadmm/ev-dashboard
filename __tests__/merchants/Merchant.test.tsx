import React from 'react';
import { act, render } from '@testing-library/react';
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';

import '../../__mocks__/react-i18next.mock';
import '../../__mocks__/matchMedia.mock';
import '../../__mocks__/i18nextMock';

import Merchant from "../../src/pages/merchants/Merchant/Merchant";

const mockMerchant = [
  {
    id: 0,
    number_stations: 0,
    active_stations: 0,
    inactive_stations: 0,
    created_by: "test",
    city_name: "test",
    country_name: "test",
    district_name: "test",
    created_at: "2019-08-24T14:15:22Z",
    name: "test",
    legal_name: "test",
    rate: "test",
    agreement_number: "test",
    address: "test",
    phone: "test",
    email: "user@example.com",
    active: true
  }
];

beforeAll(() => {
  process.env.NODE_ENV = 'development';
  jest.mock("~/shared/api/store", () => ({
    merchantStore: {
      getMerchantDetail: jest.fn(() => Promise.resolve(mockMerchant)),
      merchantDetail: mockMerchant,
    },
  }));
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: key => key })
}));

jest.mock('~/shared/utils/config', () => ({
  apiURL: 'development',
}));

afterEach(() => {
  jest.restoreAllMocks();
});

describe('Create Merchant UI Component', () => {
  test('Render component toMatchSnapshot()', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Merchant />
      </BrowserRouter>
    );
    await act(async () => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
