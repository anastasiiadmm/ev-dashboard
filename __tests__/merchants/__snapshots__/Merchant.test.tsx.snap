// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Merchant UI Component Render component toMatchSnapshot() 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-live="polite"
    class="ant-spin ant-spin-spinning spin css-dev-only-do-not-override-xu9wm8"
  >
    <span
      class="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
      <i
        class="ant-spin-dot-item"
      />
    </span>
  </div>
</DocumentFragment>
`;
